addons/project_timesheet_holidays/models/hr_employee.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+class Employee(models.Model):
+    _inherit = 'hr.employee'
+
+    @api.model
+    def create(self, vals):
+        result = super(Employee, self).create(vals)
+
+        # We need to create timesheet entries for the global time off that are already created
+        # and are planned for after this employee creation date
+
+        # First we look for the global time off that are already planned after today
+        global_leaves = self.env['resource.calendar.leaves'].search([
+            ('date_from', '>=', fields.Date.today()),
+            ('id', 'in', result.resource_calendar_id.global_leave_ids.ids)])
+        vals_list = []
+
+        # For each of them we create the corresponding timesheet entries for the employee
+        for global_time_off in global_leaves:
+            work_hours_data = global_time_off._work_time_per_day()
```

Comment:
Note for @IT-Ideas 

This method should be refactored to work in batch.

```python
    def _work_time_per_day(self):
        self.ensure_one()
        if not self.date_from.tzinfo:
            date_from = self.date_from.replace(tzinfo=utc)
        if not self.date_to.tzinfo:
            date_to = self.date_to.replace(tzinfo=utc)
        work_hours_intervals = self.calendar_id._attendance_intervals_batch(
            date_from,
            date_to)

        work_hours_data = work_hours_intervals[self.resource_id.id]
        result = defaultdict(float)
        for start, stop, dummy in work_hours_data:
            result[start.date()] += (stop - start).total_seconds() / 3600
        result = sorted(result.items())
        return result
```

The call to _attendance_intervals_batch could be factorized by calendar.

Reply:
Hello @tivisse 
I added a commit to the branch in order to improve perf as you proposed. Is this what you expected? Thanks 🙂 

cc: @jdoutreloux